OrchardCore.Commerce/Views/TextProductAttributeField.cshtml

```diff
@@ -1,28 +1,37 @@
-@using OrchardCore.Commerce.Settings
 @using OrchardCore.Commerce.Abstractions
-
+@using OrchardCore.Commerce.Settings
 @{
     int index = Model.Index;
     var attr = Model.AttributeDescription as ProductAttributeDescription;
-    var settings = attr.Settings as TextProductAttributeFieldSettings;
+    var settings = attr?.Settings as TextProductAttributeFieldSettings;
     var id = $"{Model.IdPrefix}-attribute-{index}";
-    int optionIndex = 0;
+    var optionIndex = 0;
 }
 
-<label for="@id">@attr.Name</label>
-<input type="hidden" name="line.Attributes[@index].Key" value="@attr.PartName.@attr.Name" />
-@if (settings.PredefinedValues is null || settings.PredefinedValues.Count() == 0)
+<label for="@id">@attr?.Name</label>
+<input type="hidden" name="line.Attributes[@index].Key" value="@attr?.PartName.@attr?.Name" />
+@if (settings?.PredefinedValues is null || !settings.PredefinedValues.Any())
 {
-    <input id="@id" type="text" name="line.Attributes[@index].Value" value="@settings.DefaultValue" placeholder="@settings.Placeholder" class="form-control" />
+    <input id="@id"
+           type="text"
+           name="line.Attributes[@index].Value"
+           value="@settings?.DefaultValue"
+           placeholder="@settings?.Placeholder"
+           class="form-control" />
 }
 else if (settings.MultipleValues)
 {
     <ul>
         @foreach (string predefinedValue in settings.PredefinedValues)
```

Comment:
`string` can be `var` here and in the other loops.

Reply:
Fixed for the loops. I hope there aren't any more var-able type names in .cshtml becuase I have no idea how I'd look for them...

Reply:
I take it back, these are all necessary becuase the `settings.PredefinedValues` is `IEnumerable<object>`.
![image](https://user-images.githubusercontent.com/4007293/166081884-41eb30b1-e21d-439d-a86a-75739fe511da.png)